File: averageOfLevels.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self,root: TreeNode) :
        if not root: return []
        queue = [root]
        res = []
        while len(queue) >  0:
            length= len(queue)
            sumTemp = 0;
            for i in range(length):
                node = queue.pop(0);
                sumTemp+=node.val
                if node.left : queue.append(node.left)
                if node.right : queue.append(node.right)
            res.append(sumTemp/length)
        return res    

File: averageOfLevels.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
function averageOfLevels( root ) {
    if(!root) return [];
    let queue = [root];
    let res = [];
    while ( queue.length > 0) {
        let length = queue.length ;
        let sum = 0
        for( let i = 0; i<length; ++i) {
            let node = queue.shift() ;
            sum+=node.val ;
            if(node.left) queue.push(node.left) ;
            if(node.right) queue.push(node.right);
        }
        res.push(sum/length)
    }
    return res

}